week-3/Homework/mandatory/7-password-validator.js

```diff
@@ -1,69 +1,83 @@
 /* 
 Password Validation
-
 Write a program that should check if each password in an array 
 contains a valid password (see below for password criterias) and return 
 new array with true or false booleans.
-
 Passwords must 
 - Have at least 5 characters.
 - Have English uppercase letters (A-Z)
 - Have English lowercase letters (a-z)
 - Have numbers (0-9)
 - Have non-alphanumeric symbols ("!", "#", "$", "%", ".")
-
 Passwords must not be any previous password in the passwords array. 
-
 Example 1:
 PreviousPassword = ["fhD8!yrjj", "ttkTu.wer3", "dvyyeyY!5", "qwbfj76%", "tytT3729."];
-
 Expected Result:
 PasswordValidationResult=  [false, false, false, false, true]
-
 */
 
-function validatePasswords(passwords) {
+let allUpper = /[A-Z]/;
+let allLower = /[a-z]/;
+let allNumbers = /[0-9]/;
+let allSymbols = /[!#$%.]/;
```

Comment:
Nice work! Can you explain what these lines mean? What are you using here?

Reply:
Yes, it's the regular expression,  they test/(match )  if the passwords contains the characters that is in the expression. Thank you